src/libasr/pass/intrinsic_functions.h

```diff
@@ -2425,6 +2426,73 @@ namespace Leadz {
+            ASR::ttype_t* t1, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+        
+        char* str = ASR::down_cast<ASR::StringConstant_t>(args[0])->m_s;
+        std::transform(str, str + std::strlen(str), str, [](unsigned char c) { return std::tolower(c); });
+        return make_ConstantWithType(make_StringConstant_t, str, t1, loc);
+    }
+
+    static inline ASR::expr_t* instantiate_ToLowerCase(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_optimization_tolowercase_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("n", arg_types[0]);
+        auto result = declare(fn_name, arg_types[0], ReturnVar);
+        auto itr = declare("i", ASRUtils::TYPE(ASR::make_Integer_t(al, loc, 4)), Local);
+        
+        /*
+        function toLowerCase(str) result(result)
+            character(len=*), intent(in) :: str
+            character(len=len(str)) :: result
+            integer :: i, len
+            len = len(str)
+            result = str
+            while (i < len)
+                if (result(i:i) >= 'A' .and. result(i:i) <= 'Z') then
+                    result(i:i) = char(ichar(result(i:i)) + ichar('a') - ichar('A'))
+                end if
+                i = i + 1
+            end while
+        end function
+        */
+    
+        body.push_back(al, b.Assignment(itr, i32(1)));
+        body.push_back(al, b.Assignment(result, args[0]));
+        
+        ASR::expr_t* str_len = ASRUtils::EXPR(ASR::make_StringLen_t(al, loc, args[0], int32, nullptr));
+        ASR::expr_t* cond = iLtE(itr, str_len);
+
+        ASR::expr_t* sliced_res = ASRUtils::EXPR(ASR::make_StringItem_t(al, loc, args[0], itr, arg_types[0], nullptr));
+        ASR::expr_t* ichar_res = ASRUtils::EXPR(ASR::make_Ichar_t(al, loc, sliced_res, int32, nullptr));
+
+        ASR::expr_t* ichar_A = ASRUtils::EXPR(ASR::make_Ichar_t(al, loc,
+                                    ASRUtils::EXPR(ASR::make_StringConstant_t(al, loc,
+                                                s2c(al, "A"), arg_types[0])), int32, nullptr));
+        ASR::expr_t* ichar_Z = ASRUtils::EXPR(ASR::make_Ichar_t(al, loc,
+                                    ASRUtils::EXPR(ASR::make_StringConstant_t(al, loc,
+                                                s2c(al, "Z"), arg_types[0])), int32, nullptr));
+        ASR::expr_t* ichar_a = ASRUtils::EXPR(ASR::make_Ichar_t(al, loc,
+                                    ASRUtils::EXPR(ASR::make_StringConstant_t(al, loc,
+                                                s2c(al, "a"), arg_types[0])), int32, nullptr));
+
+        body.push_back(al, b.While(cond, {
+            b.If(And(iGtE(ichar_res, ichar_A), iLtE(ichar_res, ichar_Z)), {
+                b.Assignment(result, iSub(iAdd(ichar_res, ichar_a), ichar_A))
```

Comment:
```suggestion
                b.Assignment(result, achar(iSub(iAdd(ichar_res, ichar_a), ichar_A)))
```
This needs intrinsic `achar`.

Reply:
Opened an issue for this: https://github.com/lfortran/lfortran/issues/3557